lib/tapioca/dsl/helpers/graphql_type_helper.rb

```diff
@@ -45,6 +45,10 @@ def type_for(argument)
             else
               "T.any(#{value_types.join(", ")})"
             end
+          when GraphQL::Schema::Scalar.singleton_class
+            method = Runtime::Reflection.method_of(unwrapped_type, :coerce_input)
+            signature = Runtime::Reflection.signature_of(method)
+            signature&.return_type&.to_s || "T.untyped"
```

Comment:
[The default behaviour of `Scalar` is to just return the raw value](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/scalar.rb#L9) so I _think_ it makes more sense to return `T.untyped` here rather than the `unwrapped_type`. Happy to discuss!

Reply:
Yes, this makes sense to me. However, this needs more finesse, since `return_type` can be `Void` or `NotTyped`, neither of which make sense as a parameter type.

We need to do something similar to https://github.com/Shopify/tapioca/blob/9c9b0e730cce4f35e4478bb8dd80de66238ce456/lib/tapioca/dsl/helpers/active_record_column_type_helper.rb#L124-L125

Reply:
Ok, I added this check in a commit, and will merge when the CI is green.